runtime/connections.go

```diff
@@ -163,18 +172,34 @@ func (r *Runtime) NewCatalogService(ctx context.Context, instanceID string) (*ca
 	return catalog.NewService(catalogStore, repoStore, olapStore, registry, instanceID, r.logger, migrationMetadata, releaseFunc), nil
 }
 
+func (r *Runtime) connectorDef(inst *drivers.Instance, name string) (*runtimev1.Connector, error) {
+	for _, c := range inst.Connectors {
+		// set in instance
+		if c.Name == name {
+			return c, nil
+		}
+	}
+
+	// defined in rill.yaml
+	for _, c := range inst.ProjectConnectors {
+		if c.Name == name {
+			return c, nil
+		}
+	}
+	return nil, fmt.Errorf("connector %s doesn't exist", name)
+}
```

Comment:
Can we make this resolve the connector config correctly (calling `connectorConfig`) and set it as the `config` value on the returned object? So the callers don't need to call `connectorConfig` separately

Reply:
The `drivers.Open` take `map[string]any` but the connector configs are `map[string]string`. One tricky case is handling of `allow_host_access` which is passed directly as a bool and parsed as bool in drivers. At some point I imagined we the drivers should be able to access runtime options directly or use `mapstructure.WeakDecode` to handle such cases so that `drivers.Open` can take `map[string]string`. At that point we should be able to remove `connectorConfig` usages as well. Should we take this separately ?